utils/checkLogIds.py

```diff
@@ -80,117 +26,54 @@ def do_operation(fullpath, project_path, target_file_handle, id_database, levels
-                        status = Status.FAIL
-
-                else:
-                    #ignore operation
+                elif param == "logId":
+                    # TODO: warn?
                     pass
+                else:
+                    invalid_logid_database[location_string] = param
 
-
-            elif operation == Operation.MODIFY:
-                #no match but we need to modify
-                target_file_handle.write(line)
-
-            if not is_good(status):
-                return status
-
-    return status
-
-
-def generate_ids_main(check_only, project_root, directories_to_include, directories_or_files_to_exclude, id_database):
+def check_log_ids_main(root_directory, directories_to_include, directories_or_files_to_exclude):
     """
-    Library function to check arangodb project for valid unique log ids.
-
-    If `check_only` is set to False it will also convert macros form old style
-    logging to new the new style inserting unique ids.
+    Function to check arangodb project for valid unique log ids.
 
-        :param check_only: check for uniqueness but do not modify
-        :param project_root: path to project_root
+        :param root_directory: path to project_root
         :param directories_to_include: list of directories to include
         :param directories_or_files_to_exclude: list of directories and files to exclude
 
-        :return: status
-        :rtype: int
+        :return: dict of LogIds with locations of use
+        :rtype: dict
     """
 
-    include = [ project_root + os.path.sep + d for d in directories_to_include ]
-    exclude = [ project_root + os.path.sep + d for d in directories_or_files_to_exclude ]
+    id_database = dict()
+    invalid_logid_database = dict()
 
-    operations = [ Operation.READ ]
-    if not check_only:
-        operations.append(Operation.MODIFY)
+    include = [ root_directory + os.path.sep + d for d in directories_to_include ]
+    exclude = [ root_directory + os.path.sep + d for d in directories_or_files_to_exclude ]
 
-    for root, dirs, files in os.walk(project_root):
+    for root, dirs, files in os.walk(root_directory):
         status = Status.OK
 
         dirs[:] = [ d for d in dirs if (root + os.path.sep + d).startswith(tuple(include)) ]
         if root in exclude:
             continue
 
-        for operation in operations:
-            for filename in files:
-                fullpath = root + os.path.sep + filename
-                if fullpath in exclude:
-                    continue
-                if (filename.endswith((".cpp",".h"))):
-                    status = handle_file(fullpath, project_root, id_database, operation)
+        for filename in files:
+            fullpath = root + os.path.sep + filename
+            if fullpath in exclude:
+                continue
+            if (filename.endswith((".cpp", ".h", ".tpp"))):
+                status = check_file(fullpath, root_directory, id_database, invalid_logid_database)
```

Comment:
We now simply ignore the `status`. In fact, it's always going to be `None`, since `check_file` does not return anything.

Reply:
Yeah, good spot; I just removed any mention of status.